dali/base/dadfs.cpp

```diff
@@ -5121,16 +5124,20 @@ class CDistributedSuperFile: public CDistributedFileBase<IDistributedSuperFile>
                 if (!orderedSubFiles[i])
                     ThrowStringException(-1, "CDistributedSuperFile: SuperFile %s: missing subfile part number %d of %d", logicalName.get(), i+1, n);
             }
+            bool isRestrictedSuperfile = queryAttributes().getPropBool("@restricted");
+            containsRestrictedSubfile = false;
             // Now try to resolve them all (file/superfile)
             for (unsigned f=0; f<n; f++)
             {
                 IPropertyTree &sub = *(orderedSubFiles[f]);
                 sub.getProp("@name",subname.clear());
                 Owned<IDistributedFile> subfile;
-                subfile.setown(transaction?transaction->lookupFile(subname.str(),timeout):parent->lookup(subname.str(), udesc, false, false, false, transaction, timeout));
+                subfile.setown(transaction?transaction->lookupFile(subname.str(),timeout):parent->lookup(subname.str(), udesc, false, false, false, transaction, defaultPrivilegedUser, timeout));
                 if (!subfile.get())
                     subfile.setown(transaction?transaction->lookupSuperFile(subname.str(),timeout):parent->lookupSuperFile(subname.str(),udesc,transaction,timeout));
-
+                containsRestrictedSubfile = containsRestrictedSubfile || subfile->isRestrictedAccess();
+                if (subfile->isRestrictedAccess() != isRestrictedSuperfile)
+                    UWARNLOG("CDistributedSuperFile: SuperFile '%s' and subfile '%s' have different restricted flag", logicalName.get(), subname.str());
```

Comment:
There are 2 potential issues
1) if the superfile is marked as restricted but subfiles are not, it would be possible to create a new superfile to access the data.
2) if the subfiles are restricted but superfile isn't, it would not be possible to unrestrict the superfile with the ECLWatch
The warning is to let the users know so they are can address the issue if they wish. It won't cause any other issues.  I've commented in the jira with some questions regarding how superfiles and subfile restrictions should be handled.

Reply:
My comment from JIRA:
```
My feeling is that it should allow mixed, but reading a super that has any restricted subs should only be possible if authorized.
In that scheme, there would be no separate super restricted property, it would be derived at super load time, or when you add / remove a sub.
```

Reply:
I think I agree with Jake, but it can be revisited once this PR is merged